src/DockWidget.cpp

```diff
@@ -516,12 +516,36 @@ bool CDockWidget::event(QEvent *e)
 		{
 			d->ToggleViewAction->setText(title);
 		}
+		if (d->DockArea)
+		{
+			d->DockArea->updateTitleBarVisibility();//update tabs menu
+		}
 		emit titleChanged(title);
 	}
 	return QFrame::event(e);
 }
 
 
+#ifndef QT_NO_TOOLTIP
+//============================================================================
+void CDockWidget::setTabToolTip(const QString &text)

```

Comment:
Could you elaborate what is then idea or intention of the setTabToolTip() function. What is your use case?

Reply:
You can use the standard CDockWidget :: setToolTip function; it will set a tooltip for the entire dockwidget.
The CDockWidget :: setTabToolTip method sets the hint only for the tab title heading and menu.

![tooltip](https://user-images.githubusercontent.com/2004184/51621917-5890a100-1f57-11e9-83d4-9fc29abbd704.png)

Reply:
Ok, and what is the use case for this feature? Why do you need it or what do you want to do in your application?

Normally the tool tip of the tab should show the name of the tab (the name of the dock widget). Because if you have a lot of tabs, then the name may be elided but the tool tip always provides the full name of the tab. That means, even if the tab is elided you can still read the name of the tab with the help of the tooltip. So I do not think it is a good idea to change the tooltip of the tab.

Reply:
In my application, the main information about the name of the device is written in the header. The pop-up tooltip prompt shows the extended information with the device’s IP address.
I do not agree that the name of the tab and the tooltip must match, this contradicts the very concept of pop-up tooltips.
However, I must note that if you do not use CDockWidget :: setTabToolTip, then the functionality of the tooltips will not change, and the tooltips duplicate the names of the widgets.
But if (as in my cases) you need to change the tooltip. you need to use the setTabToolTip function.
I am a shield that programmers need to be given the freedom to choose to set tooltips.

Reply:
Ok, I understand your use case now. I tested your pull request and I'm almost ready to merge it. Only two final questions left. The first one: why do you need to mark the tabs menu outdated? Normally, if you assign a tab tooltip right after creation of the dock widget, then this should work. Marking the tabs menu outdated is only required, if you would like to change the tooltip of the tab dynamically during runtime - is this your intention?

And the second is more a request than a question. Could you please use the setTabToolTip() function in the demo for one single DockWidget - i.e. for the first Calander dock widget just to show how it works and to have some kind of "test".

Thank you

Reply:
Thanks for reviewing the code.
We mark the menu for updating in the following situation.
1. create widgets
2. open the menu (items are created)
3. do something ...
4. set a new taboo name or a new tooltip.
5. open the menu and see the old name and the old tooltip in the menu.
Therefore, when we set a new hint or a new tab name, we need to set a mark on the re-creation of the menu.

Thank!

Reply:
Yes, set the menu update label to dynamically change the tooltip and the title name of the widget and item menu